src/retro_data_structures/formats/rule.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import annotations
+
+import enum
+import typing
+
+import construct
+
+from retro_data_structures.adapters.enum_adapter import EnumAdapter
+from retro_data_structures.base_resource import AssetType, BaseResource, Dependency
+from retro_data_structures.common_types import AssetId32, FourCC
+from retro_data_structures.construct_extensions.alignment import AlignTo
+from retro_data_structures.construct_extensions.misc import ErrorWithMessage
+
+if typing.TYPE_CHECKING:
+    from retro_data_structures.game_check import Game
+
+
+class ComparisonOperator(enum.IntEnum):
+    LessThan = 0
+    LessThanOrEqualTo = 1
+    EqualTo = 2
+    GreaterThanOrEqualTo = 3
+    GreaterThan = 4
+
+
+class ValueType(enum.IntEnum):
+    Bool = 0
+    Float = 1
+    Int32 = 2
+
+
+RuleCondition = construct.Struct(
+    id=FourCC,
+    operator=EnumAdapter(ComparisonOperator, construct.Int8ub),
+    value_type=EnumAdapter(ValueType, construct.Int8ub),
+    value=construct.Switch(
+        construct.this.value_type,
+        {
+            ValueType.Bool: construct.Int32ub,
+            ValueType.Float: construct.Float32b,
+            ValueType.Int32: construct.Int32ub,
+        },
+        default=ErrorWithMessage(lambda ctx: f"Unknown type: {ctx.value_type}"),
+    ),
+)
+
+Action = construct.Struct(
+    id=FourCC,
+    properties=construct.PrefixedArray(construct.Int8ub, construct.Int32ub),
```

Comment:
![Screenshot_20250410_205841_Discord](https://github.com/user-attachments/assets/a22b7587-1f56-47a7-b4f8-93b922133a03)
might be more like... 
```py
def rebuild_action_count(ctx):
    if ctx.prop3 != 0:
        return 3
    if ctx.prop2 != 1:
        return 2
    return 1
```

```py
count=construct.Rebuild(construct.Int8ub, rebuild_action_count),
prop1=construct.Float32b,
prop2=construct.If(construct.this.count >= 2, construct.Int32ub),
prop3=construct.If(construct.this.count >= 3, construct.Int32ub),
```
?